packages/nuxt/src/components/scan.ts

```diff
@@ -29,6 +25,11 @@ export async function scanComponents (dirs: ComponentsDir[], srcDir: string): Pr
   // All scanned paths
   const scannedPaths: string[] = []
 
+  const ISLAND_RE = /\.island(?:\.global)?$/
+  const GLOBAL_RE = /\.global(?:\.island)?$/
+  const COMPONENT_MODE_RE = /(?<=\.)(client|server)(\.global|\.island)*$/
+  const MODE_REPLACEMENT_RE = /(\.(client|server))?(\.global|\.island)*$/
+
```

Comment:
This has been made based on the assumption end-developers can't call scanComponents on their own via the triggering hook. From my testing, that appeared to not work. If I am wrong, this will need to be reverted